src/shogun/lib/SGMatrix.cpp

```diff
@@ -364,7 +366,8 @@ template <class T>
 void SGMatrix<T>::transpose_matrix(T*& matrix, int32_t& num_feat, int32_t& num_vec)
```

Comment:
IMO, we should remove this method. Not necessary and slow compared to passing flags to linalg/eigen

Reply:
We can get rid of this, and replace its uses with linalg::transpose_matrix, and in another PR refactor them.

Reply:
As said, I think this is an operation that shouldn’t be offered. People we it mindlessly. With the current linalg, we use flags in the operations. In some expression template approach this would be stored as a flag(like in eigen)

Reply:
I removed `transpose_matrix` operation. It was mainly used in unit-tests and documentation, so I changed those to linalg::transpose_matrix. NewtonSVM used it, but it was easy to refactor.